### ISSUE TITLE
NullPointerException when Weld Probe is enabled

### ISSUE THREAD

**Issue by ljnelson** [oracle/helidon#1004]
A paraphrase of a user report indicates that supplying `-Dorg.jboss.weld.development=true` to a Helidon MP application results (ultimately) in a `NullPointerException`:
```
java.lang.NullPointerException
	at io.helidon.microprofile.config.cdi.ConfigCdiExtension.harvestConfigPropertyInjectionPointsFromEnabledObserverMethod(ConfigCdiExtension.java:168)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	at org.jboss.weld.injection.MethodInvocationStrategy$SpecialParamPlusBeanManagerStrategy.invoke(MethodInvocationStrategy.java:187)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
	at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
	at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
	at org.jboss.weld.util.Observers.notify(Observers.java:166)
	at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
	at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:177)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:171)
	at org.jboss.weld.bootstrap.events.AbstractContainerEvent.fire(AbstractContainerEvent.java:53)
	at org.jboss.weld.bootstrap.events.AbstractDefinitionContainerEvent.fire(AbstractDefinitionContainerEvent.java:44)
	at org.jboss.weld.bootstrap.events.ProcessObserverMethodImpl.fire(ProcessObserverMethodImpl.java:49)
	at org.jboss.weld.bootstrap.events.ProcessSyntheticObserverMethodImpl.fire(ProcessSyntheticObserverMethodImpl.java:36)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessObserverMethod(ContainerLifecycleEvents.java:313)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessObserverMethod(ContainerLifecycleEvents.java:306)
	at org.jboss.weld.bootstrap.events.AfterBeanDiscoveryImpl.processObserverRegistration(AfterBeanDiscoveryImpl.java:272)
	at org.jboss.weld.bootstrap.events.AfterBeanDiscoveryImpl.finish(AfterBeanDiscoveryImpl.java:177)
	at org.jboss.weld.bootstrap.events.AfterBeanDiscoveryImpl.fire(AfterBeanDiscoveryImpl.java:76)
	at org.jboss.weld.bootstrap.WeldStartup.deployBeans(WeldStartup.java:462)
	at org.jboss.weld.bootstrap.WeldBootstrap.deployBeans(WeldBootstrap.java:86)
	at org.jboss.weld.environment.se.Weld.initialize(Weld.java:803)
	at org.jboss.weld.environment.se.Weld.initialize(Weld.java:176)
	at io.helidon.microprofile.server.Server$Builder.createContainer(Server.java:325)
	at io.helidon.microprofile.server.Server$Builder.lambda$doBuild$0(Server.java:229)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at io.helidon.microprofile.server.Server$Builder.doBuild(Server.java:227)
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:118)
	at io.helidon.microprofile.server.Server$Builder.build(Server.java:202)
	at io.helidon.microprofile.server.Server.create(Server.java:94)
	at io.helidon.microprofile.server.Main.main(Main.java:58)

	at org.jboss.weld.bootstrap.events.AbstractDefinitionContainerEvent.fire(AbstractDefinitionContainerEvent.java:46)
	at org.jboss.weld.bootstrap.events.ProcessObserverMethodImpl.fire(ProcessObserverMethodImpl.java:49)
	at org.jboss.weld.bootstrap.events.ProcessSyntheticObserverMethodImpl.fire(ProcessSyntheticObserverMethodImpl.java:36)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessObserverMethod(ContainerLifecycleEvents.java:313)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessObserverMethod(ContainerLifecycleEvents.java:306)
	at org.jboss.weld.bootstrap.events.AfterBeanDiscoveryImpl.processObserverRegistration(AfterBeanDiscoveryImpl.java:272)
	at org.jboss.weld.bootstrap.events.AfterBeanDiscoveryImpl.finish(AfterBeanDiscoveryImpl.java:177)
	... 13 more
	Suppressed: java.lang.NullPointerException
		at io.helidon.microprofile.config.cdi.ConfigCdiExtension.harvestConfigPropertyInjectionPointsFromEnabledObserverMethod(ConfigCdiExtension.java:168)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
		at org.jboss.weld.injection.MethodInvocationStrategy$SpecialParamPlusBeanManagerStrategy.invoke(MethodInvocationStrategy.java:187)
		at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
		at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
		at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
		at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
		at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
		at org.jboss.weld.util.Observers.notify(Observers.java:166)
		at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
		at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
		at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:177)
		at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:171)
		at org.jboss.weld.bootstrap.events.AbstractContainerEvent.fire(AbstractContainerEvent.java:53)
		at org.jboss.weld.bootstrap.events.AbstractDefinitionContainerEvent.fire(AbstractDefinitionContainerEvent.java:44)
```
The stack is deliberately large here to also capture the fact that the thrown exception is also suppressed in case that ends up being relevant.

I'll fix this shortly; I believe as of this writing it should be relatively straightforward.

**Comment by ljnelson** [comment]
At first glance, the issue is that the return value from a `ProcessObserverMethod` "event's" `getAnnotatedMethod()` method is `null`.  That seems wrong, but the [javadoc does mention that in the case of synthetic observer methods the return value cannot be trusted (!)](https://docs.jboss.org/cdi/api/2.0/javax/enterprise/inject/spi/ProcessObserverMethod.html#getAnnotatedMethod--).  Perhaps that is what is going on here.

**Comment by ljnelson** [comment]
Avoiding this `NullPointerException` unlocks a couple of other issues: some of the classes in `jta-cdi` are unproxyable due to private constructors.  Somewhat surprising that Weld can't detect this earlier but here we are.  I'll fix these as well.

**Comment by ljnelson** [comment]
Next up, those fixes work fine.  Development mode ("Probe") comes up.  But of course Helidon is not a Servlet environment so following any Servlet-oriented instructions about launching web UIs that are supposed to show you Weld probe data won't work.  Investigating further to see if there's an easy workaround.

**Comment by ljnelson** [comment]
It looks like the Weld Probe stuff is inextricably linked to Servlet (see https://github.com/weld/core/blob/3.1.1.Final/probe/core/src/main/java/org/jboss/weld/probe/ProbeFilter.java).

For SE integration, Weld recommends you expose the data via JMX, which should work in this case.  See (for example) http://weld.cdi-spec.org/news/2015/11/10/weld-probe-jmx/.

I'm going to apply the fixes described above and close this issue.  Then I'll look to see if it is possible to somehow integrate the probe filter in terms of JAX-RS application semantics only, but that might take a while.

**Comment by ljnelson** [comment]
Fixed with #1005.
